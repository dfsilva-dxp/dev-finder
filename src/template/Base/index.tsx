import { Outlet } from "react-router-dom";

import { Header } from "components";

import { Container, Main } from "./styles";

export default function Base() {
  return (
    <Main>
      <Container>
        <Header />
        <Outlet />
      </Container>
    </Main>
  );
}
