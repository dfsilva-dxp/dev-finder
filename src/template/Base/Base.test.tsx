import { screen } from "@testing-library/react";

import { renderWithTheme } from "utils/test/helpers";

import { Base } from "template";

jest.mock("components/Header", () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <header />;
    }
  };
});

describe("<Base/>", () => {
  it("should render header and content", () => {
    renderWithTheme(<Base />);

    const header = screen.getByRole("banner");

    expect(header).toBeInTheDocument();
    expect(header.parentElement).toHaveStyle({
      "max-width": "71.25rem",
      padding: "2.5rem 1rem"
    });
  });
});
