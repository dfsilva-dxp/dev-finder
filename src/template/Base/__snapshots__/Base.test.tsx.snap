// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base/> should render header and content 1`] = `
.c3 {
  background-color: transparent;
  border: 0;
  border-radius: 0.25rem;
  box-shadow: none;
  font-size: 0.75rem;
  font-weight: 500;
  color: #FFFFFF;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.5rem;
}

.c3:hover,
.c3:focus {
  outline: 2px solid #0178FF;
  outline-offset: 0.125rem;
}

.c2 {
  padding: 1rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 h1 {
  font-size: 1.125rem;
  font-weight: 500;
}

.c0 {
  width: 100%;
  height: 100vh;
}

.c1 {
  width: 100%;
  max-width: 67.5rem;
  height: 100%;
  margin: 0 auto;
  padding: 2.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2rem;
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <h1>
          devFinder
        </h1>
        <button
          class="c3"
        >
          Dark
          <svg
            fill="currentColor"
            height="18"
            viewBox="0 0 256 256"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <circle
              cx="128"
              cy="128"
              fill="none"
              r="60"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="128"
              x2="128"
              y1="36"
              y2="28"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="62.9"
              x2="57.3"
              y1="62.9"
              y2="57.3"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="36"
              x2="28"
              y1="128"
              y2="128"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="62.9"
              x2="57.3"
              y1="193.1"
              y2="198.7"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="128"
              x2="128"
              y1="220"
              y2="228"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="193.1"
              x2="198.7"
              y1="193.1"
              y2="198.7"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="220"
              x2="228"
              y1="128"
              y2="128"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="193.1"
              x2="198.7"
              y1="62.9"
              y2="57.3"
            />
          </svg>
        </button>
      </header>
    </div>
  </main>
</div>
`;
